lib/helpers/model/discriminator.js

```diff
@@ -34,7 +35,7 @@ module.exports = function discriminator(model, name, schema, tiedValue, applyPlu
 
   if (applyPlugins) {
     const applyPluginsToDiscriminators = get(model.base,
-      'options.applyPluginsToDiscriminators', false) || !mergeHooks;
+      'options.applyPluginsToDiscriminators', false) || !mergeHooks || !mergePlugins;
```

Comment:
I'm not a fan of how `mergePlugins` applies here. Seems a little strange that `mergePlugins` also effectively disables `applyPlugins`.

Maybe instead we reuse `applyPlugins` instead of adding a new option?